### DISCUSSION TITLE
Efficient approaches for matrix inverse operation

### DISCUSSION THREAD

**Discussion by faze-geek** [sympy/sympy#25476] [General]
This is not an issue, but a doubt. 
I am implementing a matrix method as follows : 
```
s = Symbol('s')
I = eye(A.shape[0])
G = C*(s*I - A).inv()*B + D)          # A,B,C,D are sympy matrices
```
But I'm not sure using `inv()` is good practice. 
Are there any more efficient methods to calculate the matrix `G` in Sympy ? Or maybe it could be done manually using LUdecomposition(). Kindly let me know any ways which come to mind. Thanks !

**Comment by oscarbenjamin** [comment]
I've converted this to a discussion rather than an issue. You need to say something more like what sort of expressions are in the matrices, what are their shapes and sizes etc. Ideally just show some simple code that makes the matrices (or at least something similar to your actual problem).

**Reply by faze-geek** [reply]
What I am trying to implement is this - https://lpsa.swarthmore.edu/Representations/SysRepTransformations/TF2SS.html
The matrices can consist purely of numbers or can be symbolic with a complex variable.
Consider a model with **n** states, **u** inputs and **y** outputs:
Then the shapes will be -
```
A : n*n
B : n*u
C : y*x
D : y*u
```

**Reply by oscarbenjamin** [reply]
The best approach really depends a lot on what kinds of expressions you have. See e.g. gh-25452 for discussions about performance of matrix inverse.

**Comment by sylee957** [comment]
Usually, using `inv` means finding a reusable matrix that solves any instance of `B`.
If you don't want to reuse the result of `(s*I - A).inv()`, and only use once for specific instance of `B`
you'd better try using `.rref` or `.solve`. 
`.solve` is the easiest interface `(s*I - A).solve(B)`

**Reply by faze-geek** [reply]
Thanks for the reply !
